src/hol_extension_context.ts

```diff
@@ -343,15 +343,29 @@ export class HOLExtensionContext {
      * Send selection to the terminal; preprocess to find `open` and `load`
      * calls.
      */
-    sendSelection(editor: vscode.TextEditor) {
+    sendSelection(editor: vscode.TextEditor, selection: vscode.Selection | undefined) {
         if (!this.isActive()) {
             return;
         }
 
-        let text = getSelection(editor);
+        let text;
+        if (!selection) {
+            text = getSelection(editor);
+        } else {
+            text = editor.document.getText(selection);
+        }
         text = processOpens(text);
 
-        this.holTerminal!.sendRaw(`${text};\n`);
+        const locPragma = positionToLocationPragma(editor.selection.anchor);
+        const trace = '"show_typecheck_errors"';
+        const data = [
+            'let val old = Feedback.current_trace ', trace,
+            '    val _ = Feedback.set_trace ', trace, ' 0 in (',
+            locPragma, ') before Feedback.set_trace ', trace, ' old end;',
+            `${text};`
+        ].join('');
```

Comment:
I am not thrilled that the function does different things based on optional arguments. Perhaps we can leave `sendSelection`s signature as-is and lift this bit of code into a function that adds the location trace? For example:
```ts
function addLocationPragma(text: string, position: vscode.Position) { 
    // ... 
}
```
And then add
```ts
    // ...
    text = processOpens(text);
    text = addLocationPragma(text, /* some position */);
```
wherever needed?

Reply:
Sounds good